### ISSUE TITLE
dApps - Aave - User does not have enough assets to supply a token shouldn't be able to complete the transaction

### ISSUE THREAD

**Issue by virginiabalducci** [status-im/status-desktop#15994]
# Bug Report

## Steps to reproduce

1. Log in with a user who has some balance
2. Connect your wallet to https://app.aave.com/
3. Click on Supply 
4. Select an amount that is higher of the total balance of the token on your wallet
5. Click approve on the dApp site
6. On the Status app, click "Sign" button when the Sign request pops up
7. Pop up closes and a notification shows "transaction accepted"
8. On the dApp side, an error message shows.

#### Expected behavior
User should not be able to proceed if they don't have enough assets to complete the transaction

#### Actual behavior
User can sign a transaction for which they do not have enough funds
When signing the transaction, a success notification shows
On the Activity the transaction can be seen as 0 

https://etherscan.io/tx/0xd2dd83d0342c47090d48da1e5bf70de759ff0544c03b3fe77fab856aa3a08866 shows the activity and that a fee was paid.
Wallet 0x32e12fd7c2c7984dd6de153691d901445fc8aa07
![Screenshot 2024-08-05 at 9 20 19 PM](https://github.com/user-attachments/assets/f0e559d1-2ef0-4654-8ad8-6de888965772)

![Screenshot 2024-08-05 at 9 20 31 PM](https://github.com/user-attachments/assets/363def81-10b0-4806-9965-27222491aff2)

![Screenshot 2024-08-05 at 9 29 35 PM](https://github.com/user-attachments/assets/3494c1f5-3f11-4118-aa2a-26283249832b)

On the dApp side, a correct error message shows
<img width="1072" alt="Screenshot 2024-08-05 at 9 19 57 PM" src="https://github.com/user-attachments/assets/9f4fb430-13da-4520-b87d-fe972008d75b">

https://github.com/user-attachments/assets/634d51ad-7821-449c-91ad-f05a4580fc3c

Console log
[console log aave.txt](https://github.com/user-attachments/files/16503854/console.log.aave.txt)

### Additional Information

- Status desktop version: 2.30 rc-4
- Operating System: Mac OS Sonoma

**Comment by stefandunca** [comment]
We descoped parsing the content of the transaction for v2.3 therefore we don't know the transaction is an ERC token to check against the existing balance. All we do is check fees and if the transfer is native the L1 and L2 chains.

**Comment by virginiabalducci** [comment]
It seems this was an issue coming from the dApp's end where they were letting user pick an amount that was higher than the current balance. I tried to reproduce this and the amount is either autocompleted to the max amount or if you try to input a higher amount an error message displays.
Therefore, a sign approval coming to Desktop App containing an amount higher than the current balance does not happen. Closing this issue for now.

https://github.com/user-attachments/assets/52d34f37-e643-4e1b-b70a-f9322b600827
